index.ts

```diff
@@ -7,7 +7,7 @@ const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
 export type Options = _request.CoreOptions&{property?: string, uri?: string};
 
 export type Callback =
-    (error: Error|null, response?: _request.RequestResponse,
+    (error: NodeJS.ErrnoException|null, response?: _request.RequestResponse,
```

Comment:
Are you sure this annotation is correct? We call `Callback` with `new Error()` below on line 46 which is not a `NodeJS.ErrnoException`.

Reply:
Values of type `NodeJS.ErrnoException` are a superset of those of type `Error`, so it works... would you rather I change it to `Error|NodeJS.ErrnoException`?

Reply:
This is what I [see](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/b67c928904f03d0911c99ab938b14bc2e59cad40/types/node/v7/index.d.ts#L280):

```js
    export interface ErrnoException extends Error {
        errno?: number;
        code?: string;
        path?: string;
        syscall?: string;
        stack?: string;
    }
```

You're asserting that the provided callback will be called with a `ErrnoException` or `null`. An `Error` is neither. The reason the compiler doesn't complain about this yet is because we are probably using typescript < 2.6 which uses bivariant type inference for function parameter types. I expect typescript 2.6 will start catching yet.

Or am I still misunderstanding things?

Reply:
Every field in `ErrnoException` is optional, so an `Error` can be an instance of `ErrnoException`. The behavior seems to be the same regardless of TS version.

Reply:
Ah. I get it. I think it would still be better to say `Error|ErrnoException`. It was non-intuitive to me why every `Error` is assignable to `ErrnoException` even though the latter is a subclass.

Reply:
Yep -- for intuition's sake I changed it